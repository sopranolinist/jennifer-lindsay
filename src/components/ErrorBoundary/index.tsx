import {
  createStyles,
  Theme,
  withStyles,
  WithStyles
} from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import React, { Component, ReactChild } from 'react';
import DocumentTitle from 'react-document-title';
import { getPageTitle } from '../../helpers/DataHelper';

const styles = (theme: Theme) =>
  createStyles({
    root: {
      alignItems: 'baseline',
      padding: theme.spacing(3),
      paddingTop: theme.spacing(12),
      flexGrow: 1,
      flexDirection: 'column',
      display: 'flex',
      textAlign: 'left'
    },
    errorHeader: {
      color: theme.palette.error['main'],
      marginBottom: theme.spacing(2)
    },
    errorMessage: {
      marginBottom: theme.spacing(2)
    }
  });

interface Props extends WithStyles<typeof styles> {
  children?: ReactChild;
  hasError?: string;
}
interface State {
  error: boolean;
}

class ErrorBoundary extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = { error: false };
  }

  static getDerivedStateFromError(error: Error | null) {
    return { error: true };
  }

  componentDidCatch(error: Error | null, info: object) {
    // console.log('ErrorBoundary.componentDidCatch', error, info);
  }

  render() {
    const pageTitle = 'Error';
    const { error } = this.state;
    const { children, hasError, classes } = this.props;
    if (!error && !hasError) return children;
    return (
      <div className={classes.root}>
        <DocumentTitle title={getPageTitle(pageTitle)} />
        <Typography component="h1" variant="h6" className={classes.errorHeader}>
          Error
        </Typography>
        <Typography
          component="span"
          variant="body1"
          className={classes.errorMessage}
        >
          The site has encountered an error{hasError && ':'}
        </Typography>
        {hasError && (
          <Typography
            component="span"
            variant="body1"
            className={classes.errorMessage}
          >
            {hasError}
          </Typography>
        )}
      </div>
    );
  }
}

export default withStyles(styles)(ErrorBoundary);
