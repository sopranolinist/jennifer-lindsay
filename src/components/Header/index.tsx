import AppBar from '@material-ui/core/AppBar';
import IconButton from '@material-ui/core/IconButton';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import Toolbar from '@material-ui/core/Toolbar';
import classnames from 'classnames';
import React, { useContext, useState } from 'react';
import { drawerWidth } from '../../helpers/NavHelper';
import { LayoutContext } from '../../hooks/LayoutContext';
import { CloseIcon, MenuIcon } from '../../images/Icons';
import { ReactComponent as Logomark } from '../../images/logos/Logomark.svg';
import { ReactComponent as Logotype } from '../../images/logos/Logotype.svg';
import NavPrimary from '../NavPrimary';

interface Props {}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    appBar: {
      marginLeft: drawerWidth,
      zIndex: 2000,
      [theme.breakpoints.up('xl')]: {
        borderBottom: `1px solid ${theme.palette.divider}`
      }
    },
    appBarLoggedOut: {
      marginLeft: drawerWidth,
      [theme.breakpoints.up('xl')]: {
        borderBottom: `1px solid ${theme.palette.divider}`
      }
    },
    menuButton: {
      marginRight: theme.spacing(0.5),
      [theme.breakpoints.up('md')]: {
        marginRight: theme.spacing(2)
      },
      [theme.breakpoints.up('xl')]: {
        display: 'none'
      }
    },
    logotype: {
      flexGrow: 1,
      display: 'flex'
    },
    logotypeWrapper: {
      width: 220,
      paddingTop: 10,
      [theme.breakpoints.down('sm')]: {
        display: 'none'
      }
    },
    logomarkWrapper: {
      width: 55,
      paddingTop: 10,
      [theme.breakpoints.up('md')]: {
        display: 'none'
      }
    },
    orgName: {
      color: theme.palette.text.primary,
      textTransform: 'uppercase',
      lineHeight: '2em',
      marginLeft: theme.spacing(2),
      paddingLeft: theme.spacing(2),
      borderLeft: `1px solid ${theme.palette.divider}`,
      [theme.breakpoints.up('sm')]: {
        paddingLeft: theme.spacing(3),
        marginLeft: theme.spacing(3)
      }
    },
    hasRightDrawer: {
      [theme.breakpoints.up('md')]: {
        paddingRight: drawerWidth
      }
    }
  })
);

const Header = (props: Props) => {
  const layoutContext = useContext(LayoutContext);
  const classes = useStyles();

  if (!layoutContext) {
    throw new Error('Layout context is missing');
  }
  const { rightDrawerOpen } = layoutContext;

  const [mobileOpen, setMobileOpen] = useState(false);
  const handleDrawerToggle = () => {
    setMobileOpen(!mobileOpen);
  };
  const closeDrawer = () => setMobileOpen(false);
  return (
    <div data-testid="component-header">
      <AppBar
        position="fixed"
        className={classnames(
          { [classes.appBar]: true },
          { [classes.appBarLoggedOut]: false },
          { [classes.hasRightDrawer]: rightDrawerOpen }
        )}
        elevation={1}
        color="default"
        component="div"
      >
        <Toolbar>
          <IconButton
            color="inherit"
            aria-label="Open navigation drawer"
            onClick={handleDrawerToggle}
            className={classes.menuButton}
          >
            {mobileOpen ? <CloseIcon /> : <MenuIcon />}
          </IconButton>
          <div className={classes.logotype}>
            <div className={classes.logomarkWrapper}>
              <Logomark />
            </div>
            <div className={classes.logotypeWrapper}>
              <Logotype />
            </div>
          </div>
        </Toolbar>
      </AppBar>
      
        <NavPrimary mobileOpen={mobileOpen} closeDrawer={closeDrawer} />
    </div>
  );
};
export default Header;
