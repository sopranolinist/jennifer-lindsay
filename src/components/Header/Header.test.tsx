import { render } from '@testing-library/react';
import { advanceTo, clear } from 'jest-date-mock';
import React from 'react';
import {
  authExamples,
  getFullyWrappedComponent,
  getJssWrappedComponent,
  getRouterWrappedComponent,
  mockConsole
} from '../../helpers/TestHelper';
import ErrorBoundary from '../ErrorBoundary';
import Header from './';

jest.mock('@material-ui/core/useMediaQuery', () => jest.fn());

describe('Component - Header', () => {
  beforeEach(() => {
    mockConsole();
    advanceTo(1558461651876);
  });
  afterEach(() => {
    clear();
  });

  test('should enforce auth context', async done => {
    const { container } = render(
      getJssWrappedComponent(
        getRouterWrappedComponent(
          <ErrorBoundary>
            <Header />
          </ErrorBoundary>
        )
      )
    );
    expect(container).toMatchSnapshot();
    done();
  });

  test('renders the header screen', async done => {
    const { authContextProps, authObjProps } = authExamples.signedInAuthorized;
    const { container } = render(
      getFullyWrappedComponent(<Header />, { authContextProps, authObjProps })
    );
    expect(container).toMatchSnapshot();
    done();
  });
});
