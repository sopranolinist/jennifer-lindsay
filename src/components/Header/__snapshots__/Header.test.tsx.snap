// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Header renders the header screen 1`] = `
<div>
  <div
    data-testid="component-header"
  >
    <div
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorDefault makeStyles-appBar-4 mui-fixed MuiPaper-elevation1"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="Open navigation drawer"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-6 MuiIconButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          class="makeStyles-logotype-7"
        >
          <div
            class="makeStyles-logomarkWrapper-9"
          >
            <svg>
              Logomark.svg
            </svg>
          </div>
          <div
            class="makeStyles-logotypeWrapper-8"
          >
            <svg>
              Logotype.svg
            </svg>
          </div>
        </div>
        <p
          class="MuiTypography-root makeStyles-greeting-106 MuiTypography-body1"
          id="signout-greeting"
        />
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          id="signout-button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Sign Out
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <nav
      class="NavPrimary-drawer-137"
    />
  </div>
</div>
`;

exports[`Component - Header should enforce auth context 1`] = `
<div>
  <div
    class="ErrorBoundary-root-1"
  >
    <h1
      class="MuiTypography-root ErrorBoundary-errorHeader-2 MuiTypography-h6"
    >
      Error
    </h1>
    <span
      class="MuiTypography-root ErrorBoundary-errorMessage-3 MuiTypography-body1"
    >
      The site has encountered an error
    </span>
  </div>
</div>
`;
